lib/ret/hub.ex

```diff
@@ -260,30 +260,28 @@ defmodule Ret.Hub do
 end
 
 defimpl Canada.Can, for: Ret.Account do
-  # Only creators can perform these actions
-  def can?(%Ret.Account{account_id: account_id}, action, %Ret.Hub{created_by_account_id: account_id})
-      when account_id != nil and action in [:update_hub, :kick_users, :mute_users],
-      do: true
+  def can?(%Ret.Account{} = account, :join_hub, %Ret.Hub{hub_bindings: hub_bindings})
+      when hub_bindings |> length > 0 do
+    hub_bindings |> Enum.any?(&(account |> Ret.HubBinding.member_of_channel?(&1)))
+  end
 
-  # Anyone can join an unbound hub
-  def can?(_, :join_hub, %Ret.Hub{hub_bindings: []}), do: true
+  def can?(%Ret.Account{} = account, :update_hub, %Ret.Hub{hub_bindings: hub_bindings})
+      when hub_bindings |> length > 0 do
+    hub_bindings |> Enum.any?(&(account |> Ret.HubBinding.can_manage_channel?(&1)))
```

Comment:
this should return true if the account is the owner of the room as usual as well

on that note -- we should discuss if we want to get things set up so the bot will associate the room with the account of the person who typed !hub create -- that seems mostly logical to me. it would probably need to be lazy in that the bot would have to note the user's discord id somewhere, and then set it downstream once they go through the OAuth flow. (to handle the case where they do not have an account yet)

Reply:
Um, I think I will leave this as it is until we make those decisions -- my working theory right now is that once a hub is bound to a channel, it should _never_ be unbound -- that helps cut down on the scenarios significantly. In that case, any owner/moderator of a community will always be able to edit and moderate any hub that was ever created in that community and that seems right to me. If the owner/moderator is removed from that community, they should no longer have access to those hubs, so it makes sense not to mark their Hubs account as the owner of the hub.